extensions/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/subresource/SuperPingResource.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.resteasy.test.subresource;
+
+import javax.enterprise.inject.Typed;
+import javax.ws.rs.GET;
+
+// WORKAROUND - resources are beans, without this restriction an ambiguous resolution would occur
```

Comment:
This is not very user-friendly but it's a result of the current integration design where resources are beans. We had a similar issue with filters: https://github.com/quarkusio/quarkus/issues/3369#issuecomment-520319715. I think we could try to support `InjectionTarget` in ArC but it wouldn't be trivial. So I'm wondering if we should restrict the bean types of all JAX-RS resources, ie. add `@Typed(SuperPingResource.class)` automatically to all resources. @stuartwdouglas @geoand @gsmet @patriot1burke WDYT?

Reply:
Sounds fine to me